tests/integration/targets/win_dns_delegation/tasks/main.yml

```diff
@@ -0,0 +1,27 @@
+---
+- name: Ensure AD/DNS roles are installed
+  ansible.windows.win_feature:
+    name:
+      - DNS
+    include_management_tools: true
+    include_sub_features: true
+    state: present
+  register: ensure_addns_roles
+
+- name: Reboot
+  ansible.windows.win_reboot:
+  when: ensure_addns_roles.reboot_required
+
+- name: ADD | DNS Zone Delegation
```

Comment:
This is a very bare set of tests, you should be testing all sorts of scenarios like check mode, state present/absent, idempotency. Not just the modules you should also be checking the results to make sure the module did what you told it to do.